lib/parser/lexer.rl

```diff
@@ -1769,16 +1769,21 @@ class Parser::Lexer
       };
 
       # a([1, 2])
-      e_lbrack    |
+      e_lbrack
+      => { emit(:tLBRACK)
+           fbreak; };
+
       # a()
       e_lparen
-      => { emit_table(PUNCTUATION_BEGIN)
+      => { emit(:tLPAREN)
            fbreak; };
 
-      # a(+b)
-      punctuation_begin
-      => { emit_table(PUNCTUATION_BEGIN)
```

Comment:
The rule names and behavior are made for benefit of someone who is already familiar with `parse.y`, since the most changes, and most important changes, stem from changes in upstream `parse.y`.